### PR TITLE
docs: fix doc for valueFrom fields for analysis args

### PR DESCRIPTION
**Pull request by harikrongali**
Fixes comment https://github.com/argoproj/argo-rollouts/pull/1629#discussion_r773489730 
Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-rollouts/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [ ] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [ ] I've signed my commits with [DCO](https://github.com/argoproj/argoproj)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My builds are green. Try syncing with master if they are not. 
* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo-rollouts/blob/master/USERS.md).

### CODE REVIEW

File: docs/features/analysis.md [hunk 1]
```diff
@@ -464,6 +464,38 @@ spec:
               fieldPath: status.canary.weights.canary.podTemplateHash
 ```
 
+!!! important
+    Available since v1.2
+Analysis arguments also support valueFrom for reading any Rollout fields and passing them as arguments to AnalysisTemplate.
```
Comment (huikang) [inline docs/features/analysis.md:469]: Rollout fields -> Rollout status?

File: docs/features/analysis.md [hunk 2]
```diff
@@ -464,6 +464,38 @@ spec:
...
+    canary:
+      analysis:
+        templates:
+        - templateName: args-example
+        args:
+        ...
+        - name: canary-hash
+          valueFrom:
+            fieldRef:
+              fieldPath: status.canary.weights.canary.podTemplateHash
```
Comment (huikang) [inline docs/features/analysis.md:496]: not sure if `status.canary.weights.canary.podTemplateHash` is a valid path. Could you double check? Thanks.

File: docs/features/analysis.md (hunk 1 above)
Comment (harikrongali) [reply docs/features/analysis.md:469]: it can any field including status

File: docs/features/analysis.md (hunk 2 above)
Comment (harikrongali) [reply docs/features/analysis.md:496]: agree..let me update to more useful scenario

**Review by huikang — APPROVED** [review]

**Review by jessesuen — APPROVED** [review]

**Pull request merged**